File: 2019/day1.swift
import Foundation

let inputURL: URL = URL(fileURLWithPath: "input/day1.txt")

if let inputString = try? String(contentsOf: inputURL, encoding: .utf8) {
    print("File loaded successfully\n")
	
	// Do stuff here
	
	print("Part 1")
	let arrayOfModuleMasses = inputString.split(separator: "\n")
	var fuelRequired: Float = 0
	
	for moduleMassString in arrayOfModuleMasses {
		let moduleMass: Float! = Float(moduleMassString)
		let fuelForModule = floor(moduleMass/3) - 2
		fuelRequired  += fuelForModule
	}
	
 	print("Total fuel required: " + String(fuelRequired) + "\n")
		
	print("Part 2")
	
	fuelRequired = 0
	
	for moduleMassString in arrayOfModuleMasses {
		
		var moduleMass: Float! = Float(moduleMassString)
		
		while moduleMass > 0 {
			let fuelForModule = floor(moduleMass/3) - 2
			if fuelForModule > 0 {
				fuelRequired  += fuelForModule
			}
			moduleMass = fuelForModule
		}
	}
	
 	print("Total fuel required: " + String(fuelRequired) + "\n")
		
} else { print("Error loading file\n") }

File: 2019/day1.rb
#!/usr/bin/env ruby

=begin
I decided to re-do Day 1 in Ruby as I've never used
Ruby before and wanted to get a wee taste of it.

I may use Ruby for AoC 2020!!
=end

require 'colorize'

arrayOfModuleMasses = File.read("input/day1.txt").split

if arrayOfModuleMasses.length != nil
	puts "File loaded successfully!"
	
	puts "\nPart 1".red.on_black.underline
	
	fuelRequired = 0
	
	arrayOfModuleMasses.each do |moduleMass|
		fuelforModule = (moduleMass.to_f/3).floor - 2
		fuelRequired += fuelforModule
	end
	
	puts "Total fuel required: #{fuelRequired}"
	
	puts "\nPart 2".red.on_black.underline
	
	fuelRequired = 0
	
	arrayOfModuleMasses.each do |moduleMassString|
		moduleMass = moduleMassString.to_f
		
		while moduleMass > 0 do
			fuelforModule = (moduleMass.to_f/3).floor - 2
			if fuelforModule > 0
				fuelRequired += fuelforModule
			end
			moduleMass = fuelforModule
		end
	end
	
	puts "Total fuel required: #{fuelRequired}"
	
end